lib/fs/basicfs_lstat_windows.go

```diff
@@ -77,10 +77,13 @@ func (fi *dirJunctFileInfo) IsDir() bool {
 var junctionPointModeMask os.FileMode
 
 func init() {
+	// Per https://tip.golang.org/doc/go1.23#minor_library_changes
+	// In go1.22 (and go1.23 when GODEBUG=winsymlink=0) directory junctions
+	// (aka "mount points") always have ModeSymlink set.
 	junctionPointModeMask = os.ModeSymlink
 	if version.Compare(runtime.Version(), "go1.23") >= 0 {
-		// if GODEBUG=winsymlink=0, then we are in g1.22 mode so let's check for
-		// os.ModeSymlink as well, as it can't hurt.
+		// In go1.23 directory junctions (and dedup files) always have
+		// ModeIrregular set (unless GODEBUG=winsymlink=0).
```

Comment:
That contradicts the go changelog (emphasis mine), from https://tip.golang.org/doc/go1.23#minor_library_changes:

> [...] reparse points that are *not* symlinks, Unix sockets, or *dedup* files now always have [ModeIrregular](https://tip.golang.org/pkg/os#ModeIrregular) set. [...]

My understanding of the go changelo gis that dedup files do not have mode irregular set in go 1.23. While the changed comment here says they always have it set.

I am not saying the logic here is wrong, I am still in the process of trying to understand what we do, but the comments are either contradictory or require clarification. And given this is all quite hard to understand/follow, I believe that's necessary to improve.

Reply:
@imsodin Good catch, fixed in https://github.com/syncthing/syncthing/pull/9786/commits/69f9c95028828fd0c11df73048aaf5230de431d2